src/main/java/de/picturesafe/search/elasticsearch/impl/SingleIndexElasticsearchServiceImpl.java

```diff
@@ -30,9 +30,10 @@
 
     @Autowired
     public SingleIndexElasticsearchServiceImpl(ElasticsearchService elasticsearchService,
-                                               List<IndexPresetConfiguration> indexPresetConfigurations) {
+                                               SingleIndexPresetConfigurationProvider indexPresetConfigurationProvider) {
         this.elasticsearchService = elasticsearchService;
-        this.indexPresetConfiguration = indexPresetConfigurations.get(0); // Compatibility to ElasticsearchServiceImpl
+        // Compatibility to ElasticsearchServiceImpl
+        this.indexPresetConfiguration = indexPresetConfigurationProvider.getIndexPresetConfiguration("default");
```

Comment:
This must be configurable, see also DefaultIndexConfiguration:

@Value("${elasticsearch.index.alias:default}")
private String indexAlias;

Reply:
No, it doesn't matter because the constructor demands a SingleIndexPresetConfigurationProvider. The "default" string is just to not have null in the code.

Reply:
Ok!